include/ocpp/common/message_queue.hpp

```diff
@@ -671,7 +715,7 @@ template <typename M> class MessageQueue {
                 this->in_flight->message.at(CALL_ACTION).template get<std::string>() + std::string("Response"));
             this->in_flight->promise.set_value(enhanced_message);
 
-            if (this->in_flight->isTransactionMessage()) {
+            if (is_transaction_message(this->in_flight->messageType)) {
```

Comment:
you could also write an additional overload like:
```
template <typename M> auto is_transaction_message(const ControlMessage<M>& control_message) {
  return is_transaction_message(control_message.messageType);
}
```
and with that you could write here `is_transaction_message(*this->in_flight)`;